server/src/test/kotlin/org/javacs/kt/InlayHintTest.kt

```diff
@@ -0,0 +1,154 @@
+        val regex = Regex("\\(([^)]+)\\) -> .*")
+        assertTrue(label.matches(regex))
+    }
+
+    @Test
+    fun `destrucuted declaration hints`() {
+        val predicates = listOf(
+            predicate(Position(17, 10), "Float"),
+            predicate(Position(17, 13), "Double"),
+        )
+        val result = nPredicateFilter(hints, predicates)
+        assertThat(result, hasSize(2))
+        assertThat(result, everyItem(isIn(hints)))
+    }
+
+    @Test
+    fun `should not render hint with explicit type`() {
+        val result = hints.filter {
+            it.label.left.contains("Type")
+        }
+        assertTrue(result.isEmpty())
+    }
+
+    @Test
+    fun `generic type hints`() {
+        val expected = listOf(Position(5, 13), Position(20, 7))
+
+        val result = hints.filter {
+            it.label.left.matches(Regex(": Box<([^)]+)>"))
+        }.map { it.position }
+
+        assertEquals(result.size, expected.size)
+        assertEquals(result.sortedBy { it.line }, expected.sortedBy { it.line })
+    }
+
+}
+
+class InlayHintCallableParameterTest : SingleFileTestFixture("inlayhints", "Parameters.kt") {
+
+    private val hints = languageServer.textDocumentService.inlayHint(inlayHintParams(file,  range(0, 0, 0, 0))).get()
+
+    @Test
+    fun `class parameter hints`() {
+        val predicates = listOf(
+            predicate(Position(13, 4), "x"),
+            predicate(Position(14, 4), "y"),
+            predicate(Position(15, 4), "z"),
+        )
+        val result = nPredicateFilter(hints, predicates)
+        assertThat(result, hasSize(3))
+        assertThat(result, everyItem(isIn(hints)))
+    }
+
+    @Test
+    fun `has one vararg parameter hint`() {
+        val varargHintCount = hints.filter {
+            it.label.left.contains("ints")
+        }.size
+        assertThat(varargHintCount, equalTo(1))
+    }
+
+    @Test
+    fun `mixed parameter types`(){
+        val predicates = listOf(
+            predicate(Position(17, 14), "d"),
+            predicate(Position(17, 19), "p1"),
+            predicate(Position(17, 25), "ints"),
+        )
+        val result = nPredicateFilter(hints, predicates)
+        assertThat(result, hasSize(3))
+        assertThat(result, everyItem(isIn(hints)))
+    }
+
+    @Test
+    fun `inferred lambda parameter type`() {
+        val hint = hints.filter {
+            it.label.left.contains("Int")
+        }
+        assertThat(hint, hasSize(1))
+        assertThat(hint.single().label.left, containsString("Int"))
+    }
+
+}
+
+class InlayHintChainedTest : SingleFileTestFixture("inlayhints", "ChainedMethods.kt") {
+
+    private val hints = languageServer.textDocumentService.inlayHint(inlayHintParams(file,  range(0, 0, 0, 0))).get()
+
+    @Test
+    fun `chained hints`() {
+        val predicates = listOf(
+            predicate(Position(17, 34), "List<String>"),
+            predicate(Position(18, 26), "List<Int>"),
```

Comment:
I never use inlay hints, and am not used to it in IntelliJ (I stay away from big IDEs like the plague), so my question may be stupid. Is this specific inlay hint for the intermediate map operation on line 19 in the ChainedMethods.kt test file?
```kotlin
val bar = listOf("hello", "world")
    .map { it.length * 2 }
```

I did not think there was supposed to be anything in those cases. If it makes sense, then it is probably okay 🙂 Might just also be me being tired after work and trying to understand stuff before dinner 😆

Reply:
Yup, thats the hint for `.map {}`. This is how it looks like with intellij's default hints:

<img width="236" alt="Screenshot 2023-10-11 at 20 59 53" src="https://github.com/fwcd/kotlin-language-server/assets/53015791/d5932ede-458e-4da5-8193-2628cfe10f57">